### PR TITLE
[zh-cn] Sync en updates #db54537 -> #c39ba70

### PR DESCRIPTION
**Pull request by Jinjiang**
Ref: https://github.com/vuejs/vue-loader/compare/db54537... c39ba70
Please don't mind I split my updates into 3 different pieces (also #992 & #991 ) 😉
@Justineo If some conflicts happens, I will resolve them after the other 2 PRs merged.

### CODE REVIEW

File: docs/zh-cn/features/hot-reload.md [hunk 1]
```diff
@@ -4,4 +4,18 @@
 
 ![hot-reload](http://blog.evanyou.me/images/vue-hot.gif)
 
+## 状态保留规则
+
+- 当编辑一个组件的 `<template>` 时，被编辑的组件实例将会就地被重新渲染，保留当前所有的私有状态。能够做到这一点是因为模板被编译成了无副作用的新的渲染函数。
```
Comment (Justineo) [inline docs/zh-cn/features/hot-reload.md]: 这个组件实例将就地重新渲染，并保留当前所有的私有状态

Comment (Justineo) [reply docs/zh-cn/features/hot-reload.md]: 新的无副作用的渲染函数？

File: docs/zh-cn/features/hot-reload.md [hunk 2]
```diff
@@ -4,4 +4,18 @@
...
+
+- 当编辑一个组件的 `<script>` 时，被编辑的组件实例将会就地被销毁并重新创建。(应用中其它组件的状态将会被保留) 是因为 `<script>` 可能包含带有副作用的生命周期钩子，所以将“重新加载”替换为重新渲染是必须的，这样做可以确保组件行为的一致性。这也意味着，如果你的组件带有全局副作用，则整个页面将会被重新加载。
```
Comment (Justineo) [inline docs/zh-cn/features/hot-reload.md]: 这个组件实例将被就地销毁并重新创建

File: docs/zh-cn/features/scoped-css.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# CSS 作用域
+# 有作用域的 CSS
 
 当 `<style>` 标签有 `scoped` 属性时，它的 CSS 只作用于当前组件中的元素。这类似于 Shadow DOM 中的样式封装。它有一些注意事项，但不需要任何 polyfills。它通过使用 PostCSS 来实现以下转换：
```
Comment (Justineo) [inline docs/zh-cn/features/scoped-css.md]: polyfills &rarr; polyfill

File: docs/zh-cn/features/scoped-css.md [hunk 4]
```diff
@@ -28,40 +28,50 @@
...
-    .a >>> .b {
+通过 `v-html` 创建的 DOM 内容不受作用域内的样式影响，但是你仍然可以通过深度作用选择器来为他们设置样式。
 
-    }
-    </style>
+### 还有一些要留意
 
-    <style lang="scss" scoped>
-    .a /deep/ .b {
+- **CSS 作用域不能代替 classes**。考虑到浏览器渲染各种 CSS 选择器的方式，当 `p { color: red }` 设置了作用域时 (即与特性选择器组合使用时) 会慢很多倍。如果你使用 class 或者 id 取而代之，比如 `.example { color: red }`，性能影响就会消除。你可以在[这块试验田](https://stevesouders.com/efws/css-selectors/csscreate.php)中测试它们的不同。
```
Comment (Justineo) [inline docs/zh-cn/features/scoped-css.md]: classes &rarr; class

**Comment by Jinjiang** [comment]
@kazupon I guess it can be merged right now. 😉

**Comment by kazupon** [comment]
🎉

**Pull request merged**